### PR TITLE
[ADD] website_sale_product_attachment: add attachments to products in website

### PR DESCRIPTION
**Pull request by yajo**
This module lets you publish downloadable attachments in a product page.

This is useful if you want to publish firmwares, manuals, specs, warranties,
or whatever document related to the product.

@Tecnativa TT20984 TT23657

### CODE REVIEW

**Review by chienandalu — APPROVED** [review]
Awesome! :+1:

File: website_sale_product_attachment/README.rst [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+=============
+
+Instead of configuring attachments per product, to configure products per attachment:
+
+#. You need permissions to access system settings.
+#. Go to *Settings > Dashboard > Activate the developer mode*.
+#. Go to *Settings > Technical > Database Structure > Attachments*.
+#. Pick or create one.
+#. Use the field *Attached in product templates* to choose the products.
+
```
Comment (HaraldPanten) [inline website_sale_product_attachment/README.rst:48]: Functional review 👍 

Just a minor change. I would add a little explanation that user should check "Is public document".

**Review by HaraldPanten — APPROVED** [review]

File: website_sale_product_attachment/README.rst (hunk 1 above)
Comment (pedrobaeza) [reply website_sale_product_attachment/README.rst:48]: Is that really needed? I don't think so.

Comment (HaraldPanten) [reply website_sale_product_attachment/README.rst:48]: I tried both cases. If the checkbox is set, user can see the attachments. If the chechbox is not set, attachments are not appearing.

Did I miss something?

I think that the "document downloading" functionality depends on that checkbox. Doesn't it?

Comment (yajo) [reply website_sale_product_attachment/README.rst:48]: Oh yes, good point.

**Comment by ValentinVinagre** [comment]
LGTM 👍 but... why doesn't it have tests?

File: website_sale_product_attachment/README.rst (hunk 1 above)
Comment (pedrobaeza) [reply website_sale_product_attachment/README.rst:48]: We have talked internally and the constraint is needed for avoiding leaks, so we are going to document it, but keep it.

**Comment by pedrobaeza** [comment]
@ValentinVinagre there are no backend things to test. Only a tour makes a bit sense, but we are over the budget, so no tests for now.

@Yajo please put it on the roadmap.

**Comment by ValentinVinagre** [comment]
> @ValentinVinagre there are no backend things to test. Only a tour makes a bit sense, but we are over the budget, so no tests for now.
> 
> @Yajo please put it on the roadmap.

Okay, no problem, I was curious.

**Comment by yajo** [comment]
All attended.

**Comment by HaraldPanten** [comment]
> All attended.

IMO, it's ready to merge

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**